clang/lib/Tooling/DependencyScanning/ScanAndUpdateArgs.cpp

```diff
@@ -171,6 +171,10 @@ void DepscanPrefixMapping::remapInvocationPaths(CompilerInvocation &Invocation,
   Mapper.mapInPlace(CodeGenOpts.ProfileInstrumentUsePath);
   Mapper.mapInPlace(CodeGenOpts.SampleProfileFile);
   Mapper.mapInPlace(CodeGenOpts.ProfileRemappingFile);
+
+  // Dependency output options.
+  for (auto &ExtraDep : Invocation.getDependencyOutputOpts().ExtraDeps)
```

Comment:
Could you add a comment that this is a workaround? "Proper" handling for these may be that we should treat these options as "transparent" (not affecting caching) and pass them through unchanged.